src/GovUk.Education.ExploreEducationStatistics.Data.Api/Controllers/TableBuilderController.cs

```diff
@@ -47,26 +49,55 @@ public class TableBuilderController : ControllerBase
         }
 
         [HttpPost("tablebuilder")]
+        [Produces("application/json", "text/csv")]
         [CancellationTokenTimeout(TableBuilderQuery)]
-        public Task<ActionResult<TableBuilderResultViewModel>> Query(
+        public async Task Query(
             [FromBody] ObservationQueryContext query,
             CancellationToken cancellationToken = default)
         {
-            return _tableBuilderService
+            if (Request.AcceptsCsv(exact: true))
+            {
+                Response.ContentDispositionAttachment("text/csv");
```

Comment:
Would we want to add a suggested filename to this response header at all?  Not too sure what would be good to suggest if we did - maybe the Subject name for example?  Probably not a requirement and not something necessarily to do now but just raising the idea.

Reply:
I've just given this a filename of `{releaseId}.csv`. Ideally, we might use the subject name, but we don't have it in scope and it's not really worth the effort to get it.

The frontend just sets its own filename anyway so this is a little redundant anyway.